Automattic/woocommerce-payments
# Update deposits REST API docs with decoupled deposit changes

Fixes #7847

#### Changes proposed in this Pull Request

This PR updates the deposits REST API documentation to reflect the "decoupled deposits" changes deployed on Jan 23 2024 in server PR 4500.

> [!note]
> I've uploaded a preview of this PR's version of the docs [here](https://spectacular-khapse-3d4c4d.netlify.app/#deposits).

- [x] `GET deposits/overview-all`
  * Response `deposit.next_scheduled` is no longer returned.
  * Response `balance.pending[].deposits_count` is no longer returned.
  * Response `balance.instant[].transaction_ids` is no longer returned.
- [x] `GET deposits/overview`
  * Response `next_deposit` is no longer returned.
  * Response `balance.pending.deposits_count` is no longer returned.
  * Response `instant_balance.transaction_ids` is no longer returned.
- [x] `GET deposits` 
  * `estimated` deposits are no longer returned.
  * `status_is` and `status_not_is` filter params no longer filter by `estimated`.
- [x] `GET deposits/summary`
  * `estimated` deposits are no longer returned.
  * `status_is` and `status_not_is` filter params no longer filter by `estimated`.
- [x] `GET deposits/(?P<deposit_id>\w+)`
  * `estimated` deposits are no longer returned, will return status `404` in this case.
- [x] `POST deposits`
  * Request body `transaction_ids` property is deprecated.
  * Request body now accepts optional but recommended `currency` property.
  * Implemented in PR #7828
- [x] `POST deposits/download`
  * `estimated` deposits are no longer included in CSV export.
  * `status_is` and `status_not_is` filter params no longer filter by `estimated`.

See project thread: paJDYF-aY6-p2
See discussion on removing estimated deposits: paJDYF-bD7-p2

#### Testing instructions





* Build the docs locally (requires Docker)
  * `cd docs/rest-api`
  * `./build.sh` to build the docs via a Docker container.
  * `npx serve build` to serve the built HTML at `localhost:3000`
* Alternatively, I've uploaded this PR's version of the docs manually [here](https://spectacular-khapse-3d4c4d.netlify.app/#deposits).
* Find and navigate to the **Deposits** section via the sidebar
* Ensure the changes and deprecations to the endpoint responses are correct as per the plan in server epic https://github.com/Automattic/woocommerce-payments-server/issues/4296

-------------------

- [x] Covered with tests  __N/A__
- [x] Tested on mobile __N/A__

**Post merge**



- [x] Publish the docs via the [deploy-api-docs.yml GH action](https://github.com/Automattic/woocommerce-payments/actions/workflows/deploy-api-docs.yml) to [automattic.github.io/woocommerce-payments](https://automattic.github.io/woocommerce-payments/) once merged to develop.
